app/connectors/SessionDataConnector.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.AppConfig
+import connectors.ConnectorFailureLogger.FromResultToConnectorFailureLogger
+import connectors.responses.SessionDataHttpResponse.{SessionDataResponse, SessionDataResponseReads}
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
+import uk.gov.hmrc.http.client.HttpClientV2
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SessionDataConnector @Inject()(config: AppConfig,
+                                     httpClient: HttpClientV2)(implicit ec: ExecutionContext) {
+
+  def getSessionData(implicit hc: HeaderCarrier): Future[SessionDataResponse] =
+    httpClient
+      .get(url"${config.vcSessionServiceBaseUrl}/income-tax-session-data")
+      .execute[SessionDataResponse]
+      .logFailureReason(connectorName = "SessionDataConnector")
```

Comment:
Small one, `logFailureReason` is only needed if the `HttpReads` instance that's used throws those errors, since we have a custom `HttpReads` instance for `SessionDataResponse` that doesn't throw the errors that this expects then we can probably remove this